### PR TITLE
Update precision generation rules

### PR DESCRIPTION
**Pull request by MernaMoawad**

### CODE REVIEW

**Comment by SAbdulah** [comment]
@MernaMoawad, more description about the changes, please!

File: modules/RulesPrecisions.cmake [hunk 1]
```diff
@@ -37,28 +37,28 @@ set(RP_CODEGEN         ${ECRC_CMAKE_MODULE_PATH}/precision_generator/codegen.py)
 if( NOT DEFINED RP_${CMAKE_PROJECT_NAME}_DICTIONNARY )
   message( WARNING "RulesPrecisions included before RP_${CMAKE_PROJECT_NAME}_DICTIONNARY was defined (Default is used)" )
   set(RP_${CMAKE_PROJECT_NAME}_DICTIONNARY ${ECRC_CMAKE_MODULE_PATH}/precision_generator/subs.py
-    CACHE INTERNAL "Dictionnary used for precision generation" )
+          CACHE INTERNAL "Dictionnary used for precision generation" )
```
Comment (ltaiefhatem) [inline modules/RulesPrecisions.cmake]: please remove these spaces so that they do not appear as changes

File: modules/RulesPrecisions.cmake [hunk 2]
```diff
@@ -37,28 +37,28 @@ set(RP_CODEGEN         ${ECRC_CMAKE_MODULE_PATH}/precision_generator/codegen.py)
...
 else()
-  set(RP_${CMAKE_PROJECT_NAME}_DICTIONNARY ${RP_${CMAKE_PROJECT_NAME}_DICTIONNARY}
-    CACHE INTERNAL "Dictionnary used for precision generation" )
+  set(RP_${CMAKE_PROJECT_NAME}_DICTIONNARY ${ECRC_CMAKE_MODULE_PATH}/precision_generator/subs.py
+          CACHE INTERNAL "Dictionnary used for precision generation" )
```
Comment (ltaiefhatem) [inline modules/RulesPrecisions.cmake]: spaces

**Comment by MernaMoawad** [comment]
1. Updated the syntax to be python-3 friendly.
2. Chmod command was removed because it was locking the files and then blocking the building with multiple processes(Rules Precision.cmake)
3.  Add additional support for python-2
4. expanded some variables holding commands to avoid inaccurate separators added.

File: modules/RulesPrecisions.cmake [hunk 3]
```diff
@@ -37,28 +37,28 @@ set(RP_CODEGEN         ${ECRC_CMAKE_MODULE_PATH}/precision_generator/codegen.py)
...
 endif()
-
+message("${RP_${CMAKE_PROJECT_NAME}_DICTIONNARY}")
 # Default Precisions
 # ------------------
 if( NOT DEFINED RP_${CMAKE_PROJECT_NAME}_PRECISIONS )
   message( WARNING "RulesPrecisions included before RP_${CMAKE_PROJECT_NAME}_PRECISIONS was defined (\"s;d;c;z\" is used)" )
   set(RP_${CMAKE_PROJECT_NAME}_PRECISIONS "s;d;c;z"
-    CACHE INTERNAL "Set of available precisions for the project" )
+          CACHE INTERNAL "Set of available precisions for the project" )
```
Comment (ltaiefhatem) [inline modules/RulesPrecisions.cmake]: spaces, please remove the subsequent ones too

**Review by ltaiefhatem — CHANGES_REQUESTED** [review]

**Comment by ltaiefhatem** [comment]
> * Updated the syntax to be python-3 friendly.
> * Chmod command was removed because it was locking the files and then blocking the building with multiple processes(Rules Precision.cmake)
> * Add additional support for python-2
> * expanded some variables holding commands to avoid inaccurate separators added.

I am not familiar with the changes but happy to acknowledge the changes needed to support various python versions.
Have u tested it on various python versions? on Jenkins?

File: modules/RulesPrecisions.cmake [hunk 4]
```diff
@@ -37,18 +37,18 @@ set(RP_CODEGEN         ${ECRC_CMAKE_MODULE_PATH}/precision_generator/codegen.py)
 if( NOT DEFINED RP_${CMAKE_PROJECT_NAME}_DICTIONNARY )
   message( WARNING "RulesPrecisions included before RP_${CMAKE_PROJECT_NAME}_DICTIONNARY was defined (Default is used)" )
   set(RP_${CMAKE_PROJECT_NAME}_DICTIONNARY ${ECRC_CMAKE_MODULE_PATH}/precision_generator/subs.py
-    CACHE INTERNAL "Dictionnary used for precision generation" )
+      CACHE INTERNAL "Dictionnary used for precision generation" )
 else()
-  set(RP_${CMAKE_PROJECT_NAME}_DICTIONNARY ${RP_${CMAKE_PROJECT_NAME}_DICTIONNARY}
-    CACHE INTERNAL "Dictionnary used for precision generation" )
+  set(RP_${CMAKE_PROJECT_NAME}_DICTIONNARY ${ECRC_CMAKE_MODULE_PATH}/precision_generator/subs.py
```
Comment (ltaiefhatem) [inline modules/RulesPrecisions.cmake:42]: good catch!

File: modules/RulesPrecisions.cmake [hunk 5]
```diff
@@ -37,18 +37,18 @@ set(RP_CODEGEN         ${ECRC_CMAKE_MODULE_PATH}/precision_generator/codegen.py)
 if( NOT DEFINED RP_${CMAKE_PROJECT_NAME}_DICTIONNARY )
   message( WARNING "RulesPrecisions included before RP_${CMAKE_PROJECT_NAME}_DICTIONNARY was defined (Default is used)" )
   set(RP_${CMAKE_PROJECT_NAME}_DICTIONNARY ${ECRC_CMAKE_MODULE_PATH}/precision_generator/subs.py
-    CACHE INTERNAL "Dictionnary used for precision generation" )
+      CACHE INTERNAL "Dictionnary used for precision generation" )
```
Comment (ltaiefhatem) [inline modules/RulesPrecisions.cmake]: still spaces... check the diffs to identify them

File: modules/RulesPrecisions.cmake [hunk 6]
```diff
@@ -201,48 +205,42 @@ MACRO(precisions_rules_py)
...
-
-      set(pythoncmd ${PYTHON_EXECUTABLE} ${RP_CODEGEN} -f ${CMAKE_CURRENT_SOURCE_DIR}/${_dependency_INPUT} -p ${_dependency_PREC} ${PRECISIONPP_arg} ${PRECISIONPP_prefix})
-
       string(STRIP "${_dependency_OUTPUT}" _dependency_OUTPUT)
       string(COMPARE NOTEQUAL "${_dependency_OUTPUT}" "" got_file)
 
       # Force the copy of the original files in the binary_dir
       # for VPATH compilation
       if( NOT ${CMAKE_PROJECT_NAME}_COMPILE_INPLACE )
-	set(generate_out 1)
```
Comment (ltaiefhatem) [inline modules/RulesPrecisions.cmake]: space vs tab?

File: modules/RulesPrecisions.cmake [hunk 7]
```diff
@@ -201,48 +205,42 @@ MACRO(precisions_rules_py)
...
+        set(generate_out 1)
       else( NOT ${CMAKE_PROJECT_NAME}_COMPILE_INPLACE )
-	string(COMPARE NOTEQUAL "${_dependency_OUTPUT}" "${_dependency_INPUT}" generate_out )
+        string(COMPARE NOTEQUAL "${_dependency_OUTPUT}" "${_dependency_INPUT}" generate_out )
```
Comment (ltaiefhatem) [inline modules/RulesPrecisions.cmake]: space vs tab

**Review by ltaiefhatem — CHANGES_REQUESTED** [review]

**Review by ltaiefhatem — APPROVED** [review]
Looks good

**Review by SAbdulah — APPROVED** [review]
It seems most of the changes not major. I am approving this!

**Pull request merged**